src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -306,12 +305,15 @@ public virtual void Write(ulong value)
         //
         public virtual unsafe void Write(float value)
         {
-            uint TmpValue = *(uint*)&value;
-            _buffer[0] = (byte)TmpValue;
-            _buffer[1] = (byte)(TmpValue >> 8);
-            _buffer[2] = (byte)(TmpValue >> 16);
-            _buffer[3] = (byte)(TmpValue >> 24);
-            OutStream.Write(_buffer, 0, 4);
+            Write(BitConverter.SingleToInt32Bits(value));
```

Comment:
This is changes observable behavior if somebody overridden `Write(int)` method.

Also, it is likely regressing performance since there will be an extra virtual call.